src/debug/ee/debugger.cpp

```diff
@@ -16732,6 +16732,18 @@ void *DebuggerHeap::Alloc(DWORD size)
     ret = pCanary->GetUserAddr();
 #endif
 
+#ifdef FEATURE_PAL
+    // We don't have executable heap in PAL, but we still need to allocate 
+    // executable memory, that's why have change protection level for 
+    // each allocation. 
+    // TODO: We need to look how JIT solves this problem.
+    DWORD unusedFlags;
+    if (!VirtualProtect(ret, size, PAGE_EXECUTE_READWRITE, &unusedFlags))
```

Comment:
We can't remove executable bit on free, because it can be set only on a page level, and we can't be sure that the rest of the memory allocated on this page shouldn't still be executable.